src/Sqids/SqidsEncoder.cs

```diff
@@ -76,12 +79,12 @@ public SqidsEncoder(SqidsOptions options)
 	/// <returns>A string containing the encoded ID.</returns>
 	/// <exception cref="T:System.ArgumentOutOfRangeException">If any of the integers passed is smaller than <see cref="MinValue"/> (i.e. negative) or greater than <see cref="MaxValue"/> (i.e. `int.MaxValue`).</exception>
 	/// <exception cref="T:System.OverflowException">If the decoded number overflows integer.</exception>
-	public string Encode(int number)
+	public string Encode(T number)
 	{
 		if (number < MinValue || number > MaxValue)
 			throw new ArgumentOutOfRangeException($"Encoding supports numbers between '{MinValue}' and '{MaxValue}'.");
 
-		return Encode(stackalloc[] { number }); // NOTE: We use `stackalloc` here in order not to incur the cost of allocating an array on the heap, since we know the array will only have one element, we can use `stackalloc` safely.
+		return Encode(stackalloc [] { number }); // NOTE: We use `stackalloc` here in order not to incur the cost of allocating an array on the heap, since we know the array will only have one element, we can use `stackalloc` safely.
```

Comment:
When I try this, I get an error, `Cannot take the address of, get the size of, or declare a pointer to a managed type ('T')` How did you get past this?

Reply:
Integer types are unmanaged, so on the type constraints for the class I put `where T : unmanaged // ...`